### DISCUSSION TITLE
Microservices use one prisma directory of all

### DISCUSSION THREAD

**Discussion by Daniek3d** [prisma/prisma#10571] [Q&A] [answered]
Hello,  i define this directory structure from microservices:
   erp-microservices/ 
            microservice-user
            microservice-produc
.....

The problem is that I have to define within each microservice a schema that contains only the model that the microservice needs, but when a change is made and it is sent to the db and  it deletes the other tables and leaves only the one that is defined. This is a problem since every time you have to make a change in the structure, the other models in schemas are lost in the db.

try to solve the problem in two ways:

ONE: Using a single directory in the root of the project but that when doing a model generation creates the node_modules directory in the root of the project and not within the microservice. the problem  here is that you can't have a custom param where generating generate the node_modules directory, just like custom directory of the scheme is passed.  and if it is possible to say which models to generate

SECOND: Leaving the schema in each directory of the microservice, but here is the problem when the microservices grow (currently 23) it is very tedious to have to copy the schema file in each directory, create a shell but still an error can occur.

Or prism is not for microservices ...

Thanks

**Comment by austincrim** [comment, marked as answer]
Hey @Daniek3d 👋,

I think a monorepo setup would be the best way to go here, where you have a single `schema.prisma` that is shared by the rest of your projects. I would recommend that you include all the relevant tables in the schema instead of trying to split them up.

[Here's an example](https://github.com/alitnk/nest-prisma-monorepo?ref=bestofvue.com) of a monorepo setup that should help you get started.

Let me know if you have more questions as you get into it!

**Comment by aguchux** [comment]
You may be forgetting the ideology behind microservices, every pod/microservice should have their own database, not different tables in one single database. By doing this, you are introducing a SPF in your infrastructure.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
